base_import_pdf_by_template/wizards/wizard_base_import_pdf_upload.py

```diff
@@ -73,7 +73,11 @@ def action_process(self):
         # Process + return records
         records = self.env[self.model]
         for line in self.line_ids.filtered("template_id"):
-            records += line.action_process()
+            try:
+                records += line.action_process()
+            except Exception:
```

Comment:
Code review, LGTM.

However, I think it's good practice to wrap the `try/except` block inside a `savepoint` to prevent cache issues.

```suggestion
            try:
                with self.env.cr.savepoint():
                    records += line.action_process()
            except Exception:
```

Reply:
There's no records involved here yet, as it only creates the move, and if failed, no record is created.

Reply:
> There's no records involved here yet, as it only creates the move, and if failed, no record is created.

Ok, I was just considering a scenario where a record has multiple lines:

- L1 → OK
- L2 → Error
- L3 → OK

Before the commit, if L2 failed, the transaction was fully reverted, and execution stopped. Now, execution continues, but L2 is created and not reverted. If this is the expected behavior, then it's fine.

Reply:
No, that's the question. The transaction is not reverted, as the Python crash is intercepted and no exception is raised upstream. The only possible fail is if there's a PG crash (like DB constraint or similar), but this is not the case.